core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java

```diff
@@ -327,37 +324,43 @@ private boolean putInternal(PageId pageId, byte[] page) {
           mMetaStore.addPage(pageId, new PageInfo(pageId, page.length));
         }
       }
-      if (deletePage(victim, victimPageInfo)) {
+      // Regardless enoughSpace, evicting victim to make progress
+      if (deletePage(victimPageInfo.getPageId())) {
         Metrics.BYTES_EVICTED_CACHE.mark(victimPageInfo.getPageSize());
         Metrics.PAGES_EVICTED_CACHE.mark();
       } else {
-        LOG.debug("Failed to evict page: {}", victim);
+        LOG.debug("Failed to evict page: {}", victimPageInfo.getPageId());
         Metrics.PUT_EVICTION_ERRORS.inc();
+        // something is wrong to add this page, let's remove it from meta store
```

Comment:
Change this comment to `failed to evict page, remove new page from metastore as there will not be enough space`?